### ISSUE TITLE
[BUG] Incompatible with react 18

### ISSUE THREAD

**Issue by onionhammer** [microsoft/ApplicationInsights-JS#1799]
**Description/Screenshot**
```
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^17.0.1" from @microsoft/applicationinsights-react-js@3.2.4 
npm WARN   node_modules/@microsoft/applicationinsights-react-js
npm WARN     @microsoft/applicationinsights-react-js@"^3.2.4" from the root project
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-number-format@4.9.1
npm WARN Found: react@18.0.0
npm WARN node_modules/react
npm WARN   react@"^18.0.0" from the root project
npm WARN   6 more (@testing-library/react, react-dom, @compiled/react, ...)
npm WARN 
```

**Steps to Reproduce**
1. Install latest version of react
2. Install @microsoft/applicationinsights-react-js@3.2.4

**Comment by MSNev** [comment]
#1820

**Comment by MSNev** [comment]
yes, see #1820 on how we are planning to support this and future updates (so we can better handle framework breaking changes).

As a general approach (mostly because of resourcing) we don't jump onto new versions of frameworks until it's been released for at least 6 months (as they *may* fix breaking changes).

And with the new structure we will be setup to better be able to support community contributions to speed up the process.

**Comment by MSNev** [comment]
FYI - [New React repo is now live](https://github.com/microsoft/applicationinsights-react-js) and we are planning an initial unpinning from v2.8.5 of ApplicationInsights release tomorrow (no other changes are included for this initial repo release)

This is the first step of being able to support upgrading to React 18.
